### ISSUE TITLE
emacs25Macport does not register as an OS X app on OS X

### ISSUE THREAD

**Issue by TikhonJelvis** [NixOS/nixpkgs#20514]
## Issue description

The Macport version of Emacs 25 builds fine, but doesn't show up in spotlight. Running the emacs command from the terminal only gets me a terminal version of Emacs, not a window.

I had an older Emacs 24 Macport package that worked correctly. I also just replaced it with the normal Emacs 25 package which *did* get installed properly as a Mac app, callable from Spotlight.

### Steps to reproduce

Install `emacs25Macport` on OS X and try launching emacs from Spotlight.

## Technical details

* System: OS X 10.11.6
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: 17.03pre95548.8ef3eae

**Comment by periklis** [comment]
@TikhonJelvis this is a known issue, but didn't have time to investigate a solution for it. In the meantime `open ~/.nix-profile/Application/Emacs.app` will help.

**Comment by lancelet** [comment]
I'm not sure if I'm doing something "obviously" wrong (I'm a complete Nix newbie), but on my system, the build doesn't seem to produce anything in `~/.nix-profile/Application` at all.

### Steps to reproduce
```
$ nix-env -i emacs-25.1-mac-6.0
replacing old ‘emacs-25.1-mac-6.0’
installing ‘emacs-25.1-mac-6.0’
$ ls ~/.nix-profile
Library      etc          lib          manifest.nix var
bin          include      libexec      share
```

I'm mainly wondering: is this the same for @TikhonJelvis, or is there something I'm missing like a basic configuration flag to enable building Mac app bundles?

### System details
  - macOS 10.12.1
  - Nix version: nix-env 1.11.4
  - Nixpkgs version: 17.03pre95548.8ef3eae

**Comment by TikhonJelvis** [comment]
Yes, it sounds like I had the same problem: it didn't produce an `$out/Applications` folder or put anything in it. The configure step *does* call `./configure` with Mac-specific flags (`--with-mac` and `--enable-mac-app=$$out/Applications`), but this doesn't seem to do the right thing.

I spent a few hours playing around with it yesterday, but couldn't figure out what the underlying problem was.

**Comment by sellout** [comment]
So, I am perhaps seeing a different issue, but I’ll mention it here to see.

If I install emacs via `nix-env -i emacs25Macport`, it works fine. However, if I use `emacsWithPackages`, then I only get bin/ and share/ directories in `$out`. It looks like a bug in emacs/wrapper.nix that it only copies over those particular directories.

**Comment by periklis** [comment]
hey folks, i want just to link to a similar discussion [(here)](https://github.com/Homebrew/legacy-homebrew/issues/4635) by homebrew users. It seems not to be a specific issue with nix or nixpkgs. The /nix/store paths as well as the symlinks on ~/.nix-profile/Applications are not searchable by spotlight. You can write you own Autormator script to open emacs and put it under /Applications.

**Comment by periklis** [comment]
@sellout Are you sure that `emacsWithPackages` uses the `emacs/macport-25.1.nix` derivation and not `emacs/default.nix`?

**Comment by sellout** [comment]
@periklis I do the following to use the macport instead of default:
```
(let                                                                                                                                                                                                                                                                             
   customEmacs = pkgs.emacsPackagesNg.override (super: self: {                                                                                                                                                                                                                   
     emacs = pkgs.emacs25Macport;                                                                                                                                                                                                                                                
   });                                                                                                                                                                                                                                                                           
 in                                                                                                                                                                                                                                                                              
   customEmacs.emacsWithPackages (epkgs: (with epkgs.melpaPackages; [...])))                                                                                                                                                                                                                                                                
```
And re: ~/.nix-profile/Applications, the problem is that there is no Emacs.app even there. I run a command like https://nixos.org/wiki/Nix_on_OS_X#Configuration to put the apps in Spotlight-friendly locations.

I think I found the underlying emacsWithPackages problem and reported it as #22893, but it seems @TikhonJelvis and @lancelet are running into something else (unless they’ve similarly configured emacsWithPackages).

**Comment by periklis** [comment]
@sellout Ok sounds we have a missing-wrapper-issue vor darwin-built binaries landing under Contents.

**Comment by mmahut** [comment]
Are there any updates to this issue, please?
